app/customer.py

```diff
@@ -0,0 +1,66 @@
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list[int],
+            money: int | float
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.home_location = location
+        self.location = location

```

Comment:
There is a duplication of the 'location' attribute assignment. Both 'self.home_location' and 'self.location' are assigned the same 'location' value. Consider removing one of these assignments to adhere to the DRY principle.